File: 1805-num-diff-ints-on-str/main.go
func numDifferentIntegers(word string) int {
    integers := make(map[string]bool)
	runes := ""

	for _, c := range word {
		if unicode.IsDigit(c) {
			runes = runes + string(c)
		} else if len(runes) > 0 {
			integers[strings.TrimLeft(string(runes), "0")] = true
			runes = ""
		}
	}

	if len(runes) > 0 {
		integers[strings.TrimLeft(string(runes), "0")] = true
	}

	return len(integers)
}

File: 1805-num-diff-ints-on-str/main.py
  def numDifferentIntegers(self, word: str) -> int:
        nits = {}
        runes = ""
        for s in word:
            if s.isdigit():
                runes += s
            elif len(runes) > 0:
                nits[runes.lstrip("0")] = True
                runes = ""

        if len(runes):
            nits[runes.lstrip("0")] = True

        return len(nits)
